libmultilabel/linear/metrics.py

```diff
@@ -126,6 +126,40 @@ def reset(self):
         self.num_sample = 0
 
 
+class Recall:
+    def __init__(self, num_classes: int, average: str, top_k: int):
+        """Compute the Recall@K.
+
+        Args:
+            num_classes: The number of classes.
+            average: Define the reduction that is applied over labels. Currently only "samples" is supported.
+            top_k: Consider only the top k elements for each query.
+        """
+        if average != "samples":
+            raise ValueError("unsupported average")
+
+        _check_top_k(top_k)
+
+        self.top_k = top_k
+        self.score = 0
+        self.num_sample = 0
+
+    def update(self, preds: np.ndarray, target: np.ndarray):
+        assert preds.shape == target.shape  # (batch_size, num_classes)
+        top_k_idx = np.argpartition(preds, -self.top_k)[:, -self.top_k :]
+        num_relevant = np.take_along_axis(target, top_k_idx, -1).sum(axis=-1)  # (batch_size, )
+        # zero label instances treated as 0, to be consistent with torchmetrics
+        self.score += np.nan_to_num(num_relevant / target.sum(axis=-1), nan=0.0).sum()
+        self.num_sample += preds.shape[0]
+
+    def compute(self) -> float:
+        return self.score / self.num_sample
+
+    def reset(self):
```

Comment:
According to the [documentation](https://torchmetrics.readthedocs.io/en/stable/pages/implement.html), user should not implement reset by theirselves.